main.cpp

```diff
@@ -44,33 +51,113 @@ your task:
+    std::vector<Car> cars;
+    std::vector<Motorcycle> motorcycles;
+    std::vector<SemiTruck> trucks;
+    
+    /*
+     instantiating vectors like this creates unexpected copies during the construction process:
+     
+     std::vector<std::string> strings
+     {
+        std::string("a string"),
+        std::string("another string)"
+     };
+     
+     We have not learned how to properly implement copy constructors or copy assignment operators yet at this point in the course.
+     These functions are part of the group of special member functions that all UDTs have.
+     
+     Instead we will learn about in-place construction to avoid making unexpected copies of our UDTs.
+     Unexpected copies could cause Undefined Behavior (UB)
+     The rule of 3-5-0 video will explain these special member functions and this UB in more detail.
+     
+     For now, these special member functions will be implemented using the compiler's default implementation.
+     the Vehicle class shows how to do this.
+     
+     You must implement these special member functions for all of the classes that derive from Vehicle.
+     */
+    
+    /*
+     In order to correctly construct an object in-place inside a vector, we must first reserve space for it.
+     
+     reserve how ever many cars, motorcycles, and trucks you'll create first
+     */
+    cars.reserve(4); //reserving room for 3 Car instances
+    
+    
+    /*
+     Now that we have reserved space for our UDT instances inside the vector, we can construct them in-place inside the vector.
+     emplace_back is passed the constructor arguments for your UDT.
+     
+     use the vector member function 'emplace_back' to construct your car/truck/motorcycle instances in-place
+     */
+    cars.emplace_back("Janice"); //constructing the first Car instance in-place in the cars vector
+    
+    /*
+     construct 2 more Car instances via emplace_back.
+     */
+    cars.emplace_back("Goofy"); //constructing the second Car instance in-place in the cars vector
+    cars.emplace_back("Donald"); //constructing the third Car instance in-place in the cars vector     
+         
+    
+    /*
+     now reserve and emplace_back your Trucks and Motorcyles
+     */
+    motorcycles.reserve(2); //reserving room for 3 Motorcycle instances
+    trucks.reserve(2); //reserving room for 3 Semitruck instances
+    // ----------     
+    motorcycles.emplace_back("James"); //constructing the first Car instance in-place in the cars vector
+    motorcycles.emplace_back("Bond"); //constructing the second Car instance in-place in the cars vector
+    trucks.emplace_back("The old man"); //constructing the first Semitruck instance in-place in the trucks vector
+    trucks.emplace_back("I<3SelfDrivingTrucks"); //constructing the second Semitruck instance in-place in the trucks vector
+    trucks.emplace_back("Los Pollos Hermanos"); //constructing the third Semitruck instance in-place in the trucks vector   
```

Comment:
You didn't reserve room for 3, so this causes all of the previous elements in the container to be copied.
- [ ] fix `reserve()` call for `trucks`